test/OpenTelemetry.Extensions.Tests/Trace/AutoFlushActivityProcessorTests.cs

```diff
@@ -42,7 +42,7 @@ public void AutoFlushActivityProcessor_FlushAfterLocalServerSideRootSpans_EndMat
         using var activity = source.StartActivity("name", ActivityKind.Server);
         activity.Stop();
 
-        mockExporting.Protected().Verify("OnForceFlush", Times.Once(), 5_000);
+        mockExporting.Protected().Verify("OnForceFlush", Times.AtLeastOnce(), 5_000);
```

Comment:
This test again! Such a pain 🤣 I spent a bunch of time on it a while back. [Updating moq](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/827#discussion_r1050135150) seemed to help. Is it failing only on some older runtimes now?

Reply:
This test failed for me on Net7 today. I recognized this test, I've seen it fail before, but I haven't kept a record of what platforms.

My intent with this PR is to make the test more resilient.
Actually, I have on ongoing _conflict_ with Activity-based unit tests and I've been working to make them more resilient when I see them fail.

Reply:
The branch where it failed up-to-date with the version bump from my PR? I think this change is fine but it does change the test where it might not catch an actual regression, no? Any desire on your part to rewrite it so that it doesn't use moq? 😄

Reply:
I think we're going to continue to have problems unit testing Activities because the ActivityStopped can get triggered twice.
See: https://github.com/dotnet/runtime/issues/76757 and https://github.com/Azure/azure-sdk-for-net/pull/31567

Regarding a rewrite...
I see that it's asserting the mock and not actually asserting the AutoFlush Processor.
I could change this to verify that OnForceFlush is called while exporting this specific Activity (by asserting the Activity.Name).
But this won't avoid that the Activity can be exported twice.

Reply:
I think for the sake of this test we only care to track invocations to OnForceFlush so it should be fine! Try this:

```csharp
using System;
using System.Diagnostics;
using System.Runtime.CompilerServices;
using System.Threading;
using OpenTelemetry.Trace;
using Xunit;

namespace OpenTelemetry.Extensions.Tests.Trace;

public class AutoFlushActivityProcessorTests
{
    [Fact]
    public void AutoFlushActivityProcessor_FlushAfterLocalServerSideRootSpans_EndMatchingSpan_Flush()
    {
        var processor = new ForceFlushTrackingProcessor();

        var sourceName = GetTestMethodName();

        using var provider = Sdk.CreateTracerProviderBuilder()
            .AddProcessor(processor)
            .AddAutoFlushActivityProcessor(a => a.Parent == null && (a.Kind == ActivityKind.Server || a.Kind == ActivityKind.Consumer), 5000)
            .AddSource(sourceName)
            .Build();

        using var source = new ActivitySource(sourceName);
        using var activity = source.StartActivity("name", ActivityKind.Server);
        activity.Stop();

        Assert.Equal(1, processor.ForceFlushInvocations);
    }

    [Fact]
    public void AutoFlushActivityProcessor_FlushAfterLocalServerSideRootSpans_EndNonMatchingSpan_DoesNothing()
    {
        var processor = new ForceFlushTrackingProcessor();

        var sourceName = GetTestMethodName();

        using var provider = Sdk.CreateTracerProviderBuilder()
            .AddProcessor(processor)
            .AddAutoFlushActivityProcessor(a => a.Parent == null && (a.Kind == ActivityKind.Server || a.Kind == ActivityKind.Consumer))
            .AddSource(sourceName)
            .Build();

        using var source = new ActivitySource(sourceName);
        using var activity = source.StartActivity("name", ActivityKind.Client);
        activity.Stop();

        Assert.Equal(0, processor.ForceFlushInvocations);
    }

    [Fact]
    public void AutoFlushActivityProcessor_PredicateThrows_DoesNothing()
    {
        var processor = new ForceFlushTrackingProcessor();

        var sourceName = GetTestMethodName();

        using var provider = Sdk.CreateTracerProviderBuilder()
            .AddProcessor(processor)
            .AddAutoFlushActivityProcessor(_ => throw new Exception("Predicate throws an exception."))
            .AddSource(sourceName)
            .Build();

        using var source = new ActivitySource(sourceName);
        using var activity = source.StartActivity("name", ActivityKind.Server);
        activity.Stop();

        Assert.Equal(0, processor.ForceFlushInvocations);
    }

    private static string GetTestMethodName([CallerMemberName] string callingMethodName = "")
    {
        return callingMethodName;
    }

    private sealed class ForceFlushTrackingProcessor : BaseProcessor<Activity>
    {
        private int forceFlushInvocations;

        public int ForceFlushInvocations => this.forceFlushInvocations;

        protected override bool OnForceFlush(int timeoutMilliseconds)
        {
            Interlocked.Increment(ref this.forceFlushInvocations);

            return true;
        }
    }
}
```